### DISCUSSION TITLE
How to use Fourier functionality?

### DISCUSSION THREAD

**Discussion by VikingExplorer** [mathnet/mathnet-numerics#1074] [Q&A]
Does anyone know how to use the Fourier functionality?

1. I would like to create input data that is composed of several different sinusoids of different frequencies. 
2. Then call FFT to get the coefficients. 
3. Make sure all other frequencies have zero for coefficients.

I created an issue so that the documentation can be improved regarding Fourier: https://github.com/mathnet/mathnet-numerics/issues/1073

**Comment by strMikhailPotapenko** [comment]
Hello @VikingExplorer 

I know you asked this question a long time ago so you are likely to have either figured out the answer yourself or have given up. Nonetheless, here is a code snippet showing one way to carry out your steps. It can be modified to use different units for frequency, generate the sinusoids in a different way, etc.

```
private static Complex[] GenerateDataAndCalculateFourierSpectrum()
{
    int signalLength = 1000;

    double[] frequencies = new double[] { 100, 250, 475 };
    Complex[][] signals = frequencies.Select(frequency => GenerateSignal(frequency, signalLength)).ToArray();
    Matrix<Complex> signalMatrix = Matrix.Build.DenseOfRowArrays(signals);
    Complex[] resultantSignal = signalMatrix.ColumnSums().ToArray();
    Fourier.Forward(resultantSignal);

    return resultantSignal;
}
private static Complex[] GenerateSignal(double omega, int signalLength)
{
    double[] phases = Generate.LinearSpaced(signalLength, 0, 2 * Math.PI);
    Complex[] signal = phases.Select(phase => Complex.FromPolarCoordinates(1, (float)(omega * phase))).ToArray();
    return signal;
}
```

Here is a plot of the magnitudes of the coefficients. As you can see, the peaks match the frequency numbers given.

![image](https://github.com/mathnet/mathnet-numerics/assets/149536475/481f09cb-50fc-4ee0-b7bd-abd5877aba70)

Hope this helps!

-strMikhailPotapenko
